protocol/src/main/java/net/md_5/bungee/protocol/packet/Team.java

```diff
@@ -123,6 +126,105 @@ public void write(ByteBuf buf, ProtocolConstants.Direction direction, int protoc
         }
     }
 
+    @SuppressWarnings("unchecked")
+    public Either<String, BaseComponent> getDisplayName()
```

Comment:
All these new methods in the packets are basically just for compatibility right?

Given packets aren't API, I think if we clean up with a ChatDeserializable and make that class as helpful as possible, we shouldn't need all this getter/setter boilerplate in each packet

Reply:
If packets are not API, we can get rid of the getter/setter/alternative constructors - would need a few adjustments inside the proxy tho.

I wasn't 100% sure how exactly this is being handled, so I jsut added these. Later on, these methods also allowed for a server to test this patch easier (and eventually finding the [mistake](https://github.com/SpigotMC/BungeeCord/pull/3634#issuecomment-2025991092) I made).

Reply:
Removed all compatiblity.